test/integration/esClient.js

```diff
@@ -0,0 +1,64 @@
+const chai = require('chai');
+const esClient = require('../../app/lib/esClient');
+const utils = require('./testUtils');
+
+const expect = chai.expect;
+
+describe('esClient', function test() {
+  this.timeout(utils.maxWaitTimeMs);
+
+  before((done) => {
+    utils.waitForSiteReady(done);
+  });
+
+  describe('getQuery', () => {
+    const location = { longitude: -1.46519099452929, latitude: 54.0095586395326 };
+    it('should return pharmacies as an array', (done) => {
+      esClient.getPharmacies(location).then((pharmacies) => {
+        expect(pharmacies).to.be.an('array');
+        expect(pharmacies.length).to.be.greaterThan(0);
+        expect(pharmacies[0].name).to.exist;
+        done();
+      }).catch(done);
+    });
+
+    it('number of results should default to 2500', (done) => {
+      esClient.getPharmacies(location).then((pharmacies) => {
+        expect(pharmacies).to.be.an('array');
+        expect(pharmacies.length).to.be.lessThan(2501);
+        expect(pharmacies.length).to.equal(529);
+        done();
+      }).catch(done);
+    });
+
+    it('number of results should be parameterised', (done) => {
+      esClient.getPharmacies(location, 10, 7).then((pharmacies) => {
+        expect(pharmacies).to.be.an('array');
+        expect(pharmacies.length).to.equal(7);
+        done();
+      }).catch(done);
+    });
+
+    it('should return pharmacies populated with distance from location', (done) => {
+      esClient.getPharmacies(location).then((pharmacies) => {
+        pharmacies.forEach(p => expect(p.dis).to.be.greaterThan(0, `Pharmacy ${p.identifier} does not have a distance`));
+        done();
+      }).catch(done);
+    });
+
+    it('reducing the radius should return fewer pharmacies', (done) => {
```

Comment:
As above, this test feels like it is testing Elasticsearch more than it is testing the code we've written. Inspecting the generated query would provide a more stable test.

Reply:
Regarding the above and similar tests, I've split the query builder out into a separate file and added unit tests for creating the query definition.

I think there is value in some integration tests that call Elasticsearch to ensure the query is valid. These are run against a snapshot of data, and are checking differences in data, rather than specific fields, so they should be fairly stable. 

The above test is the only one that would fail if data changed, i.e. all pharmacies between 1 and 5 miles away were no longer in business when the data is refreshed.